bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -610,10 +627,11 @@ impl Timeline {
         event_or_transaction_id: EventOrTransactionId,
         reason: Option<String>,
     ) -> Result<(), ClientError> {
-        self.inner
-            .redact_by_id(&(event_or_transaction_id.try_into()?), reason.as_deref())
-            .await
-            .map_err(Into::into)
+        if !self.inner.redact(&(event_or_transaction_id.try_into()?), reason.as_deref()).await? {
+            // TODO make it a hard error instead
```

Comment:
Can we please change the underlying API to return `Result<bool, Error>` at this point please? We already have a `RedactError`, it shouldn't be too much of a stretch to convert into it from the underlying handle abortion boolean.